### DISCUSSION TITLE
Show/Hide Columns within a Table

### DISCUSSION THREAD

**Discussion by brannytempy** [twilio-labs/paste#2761] [Q&A] [answered]
Hi Paste!

In Segment, we have quite a few tables that have this button that lets a user simply turn on/off certain columns. Users can select which columns they care about.

![CleanShot 2022-10-21 at 11 34 13](https://user-images.githubusercontent.com/1590723/197233987-8f20e8f3-80e7-4643-baff-78116df92a55.png)

Can we achieve this with Tables/Data Grids? Any recommendations or advice on how to approach this to replicate this functionality? [I found this discussion which is similar.](https://github.com/twilio-labs/paste/discussions/2338) Is your guidance the same here?

**Comment by nkrantz** [comment, marked as answer]
Hi @brannytempy! You can still use checkboxes to show/hide columns following the guidance in the discussion you linked. We just released our [Side Modal component](https://paste.twilio.design/components/side-modal/#about-side-modal), though, which is another good option for displaying the column filter (rather than a Popover).
